### PR TITLE
Assignment from Michal Karaba

### PR DESCRIPTION
**Pull request by calom**
Hi,

my take on the cypress and Trezor suite. 

Most of the time I had various problems running docker/docker-suite-test.sh locally where few attempts were needed to start cypress in docker. I tried to play with Docker ram setting, closing other applications but overall it felt very slow and unstable (I tested solely with docker as proposed in et2.md)

Not sure if intentional but phassphrase check is missing initial setup, at least with how assignment is written. I have written down potential steps. I tried to set it up, but test became super unstable on my local with it so I left it only in comment without initial config. 

I would be I could run the app and use instance of cypress directly which would make it much faster and stable for development but was not able build it somehow and with all the setup aimed for docker I used what you suggested and have. 

Overall it was interesting experience but felt heavy on first attempt in a sense, that I was able to run test successfully and with very little change  in the test failed on rerun. At that point I was very eager to discuss this with some potential colleague on "what is going on with it".

Cheers and see you on 3rd round for feedback,
Michal

### CODE REVIEW

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        cy.task('setupEmu');
+        cy.task('startBridge');
+    });
+
+    it('e2e test for assignment', () => {
+        // This is the check for "https://suite.trezor.io/web/ )" which is generated on local host
+        cy.url().should('include', 'localhost:8000');
+        // Passing initial stages of Trezor Suite
+        cy.passThroughInitialRun();
+
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:15]: Jeden nezbytny krok v zadani naschval neni popsany - hned po zalogovani do suite na uzivatele vyskoci modal `Select wallet type`, ktery je potreba vyresit. Test takto selze hned na zacatku.


image

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        cy.task('startEmu', { wipe: true });
+        cy.task('setupEmu');
+        cy.task('startBridge');
+    });
+
+    it('e2e test for assignment', () => {
+        // This is the check for "https://suite.trezor.io/web/ )" which is generated on local host
+        cy.url().should('include', 'localhost:8000');
+        // Passing initial stages of Trezor Suite
+        cy.passThroughInitialRun();
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:14]: Tenhle command je tady uz zbytecny a vede k potencialnimu padu testu. Staci ho udelat jednou v ramci `beforeEach` metody.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        /**
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:16]: Spolecne s predchozim resenim modalu je nutne zaradit i command `cy.discoveryShouldFinish()`, jinak vam test skonci v timeoutu.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+         * I Had various results with it but mostly it helped and test was more stable.
+         */
+        cy.getTestElement('@suite/menu/settings').should('be.visible').click();
+        cy.getTestElement('@settings/menu/device').click();
+
+        /**
+         * I have left wait here as test worked somewhat better with it although I dont fully understand why.
+         * Without that wait I had troubles clicking setting icon. Maybe my mac is overall slow with it.
+         */
+        cy.wait(2000);
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:27]: Staticky wait je az na velmi vyjimecne situace nechtenna vec a da se resit elegantnej.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        // cy.getTestElement('@suite/menu/suite-index').click();
+        /**
+         * Here maybe putting whole line 57 inside block for the box in which this phrase suppose to be would be even more precise.
+         * I saw it is possible. Feel more clear on what I intent to assert with it.
+         */
+        // cy.get('Passphrase enabled');
+
+        // Last check of renamed Trezor
+        cy.getTestElement('@menu/switch-device').click();
+        cy.get('Done!');
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:64]: Test na tomto selze, `'Done!` neni platny css selektor, takze takhle nejde pouzit. Tady jsem si predstavoval pouziti `cy.contains()` metody spolecne s pouzitim `.then()` callbacku po `cy.get()` na modal (abychom modal odrizli od zbytku aplikace, cimz zamezime druhemu stringu `Done!`, aby zkazil zaverecny assert.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        cy.wait(2000);
+        cy.getTestElement('@settings/device/label-input')
+            /**
+             * Focus with type and invoke somehow work together. On my local run, when I used only invoke('val', '')
+             * in theory it suppose to be enough, but wasn't. Once I added those other parts it was doing what it suppose to.
+             * Haven't figured out why standalone clear() nor type('{selectall}{backspace}') have not worked. I have noticed
+             * my focus was behind upper tab where Dashboard, Device and other tabs remains, but don't know what is behind it.
+             * All in all focus().clear().type() should be working but didn't.
+             */
+            .focus()
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:36]: Tohle je funkcni reseni, ale pres code review by neproslo, protoze je to prilis velky workaround.

Je to malych "chytak". Cypress se neumi vyporadat se sticky a fixed css vlastnostmi na elementech na strance (tohle je vlastnost frontendu, ne chyba testu) a vede to k tomu, ze cypress donekonecna scrolluje dolu.

Problem se da resit manualnim scrollingem, pouzitim .({ force: true }) modifikatoru anebo nami preferovane reseni pouzite v jinych testech - nastaveni vetsiho viewportu na zacatku testu -> cy.viewport(1080, 1440)

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+         * This is not implemented. Setting phassphrase settings by my self always resulted in various weird behaviors
+         * and test was very unstable. I tried it with  cy.task('setupEmu', { passphrase_protection: true }); And steps to click passphrase.
+         * I believe there is some auto config somewhere but I was not able to find it.
+         */
+        // cy.getTestElement('@suite/menu/suite-index').click();
+        /**
+         * Here maybe putting whole line 57 inside block for the box in which this phrase suppose to be would be even more precise.
+         * I saw it is possible. Feel more clear on what I intent to assert with it.
+         */
+        // cy.get('Passphrase enabled');
```
Comment (ondracja) [inline packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:60]: Bez ohledu na implementaci by tento get selhal - neni to platny css vyraz. Predstavoval jsem si tam bud kontrolu neexistence`Enable` buttonu anebo pritomnost elementu "zelena fajfka".

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts (hunk 2 above)
Comment (calom) [reply packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:14]: Ono to bolo v beforeEach, ale po skusani mnohych veci to nakoniec skoncilo tu. Chapem, ze pri viacerych testoch by to bola duplikacia. Pri danom rieseni som na to az tak neprihliadal z pohladu jedneho testu.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts (hunk 4 above)
Comment (calom) [reply packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:27]: Tomu rozumiem. Pri lokalnom behu mi elegantnejsie riesenie nefungovalo. Mal som tam nieco na styl 'wait until not present' metody ale to mi vytimeoutovalo. 

Tiez som nechapal, preco test nechcel kliknut na iconu, aj ked uz bola zobrazena. Sice v logu na tu ikonu klikol a isiel na dalsie kroky, ale realne ostal v danom okne a potom som dostal timeout, kedze test uz riesil dalsi krok, ktory sa mal riesit v novom okne.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts (hunk 6 above)
Comment (calom) [reply packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:36]: Ten viewport som myslel, ze zaberie, ale zial som prebral len zakladne rozlisenie a neskusil vyssie.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts (hunk 7 above)
Comment (calom) [reply packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:60]: Rozumiem, v celku to takto dava zmysel overit vsetky zmeny na elemente. Vidim, ze ja som sa moc sustredil na ten string zapisany v zadani a zvysok karty neriesil.

File: packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts (hunk 1 above)
Comment (calom) [reply packages/integration-tests/projects/suite-web/tests/assignment/assignment.test.ts:15]: Myslim, ze sa mi to nezobrazovalo, cize zo zaciatku som passThroughInitialRun ani nepouzival, az ked som mal passphrase true to zacalo davat zmysel. 

Ono sa mi to javilo skor tak, ze ta vybuildena appka sa po spusteny a nastaveni emulatora ma prihlasit, a refresh stranky s appkou je tam ako taky workaround po ktorom som uz bol v aplikacii Trezor Suite. To mi na zaciatku aj fungovalo.

**Pull request closed without merging**